tests/integration/self_update_test.sh

```diff
@@ -0,0 +1,213 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+include './src/lib/kwlib.sh'
+include './tests/unit/utils.sh'
+include './tests/integration/utils.sh'
+
+declare -a KW_BRANCHES=('master' 'unstable')
+declare -g HOST_KW_CLONED_PATH
+
+function oneTimeSetUp()
+{
+  local kw_repo_url='https://github.com/kworkflow/kworkflow'
+
+  HOST_KW_CLONED_PATH="$(mktemp --directory)/kworkflow"
+
+  # Clone the repository once on the host
+  git clone "$kw_repo_url" "$HOST_KW_CLONED_PATH" > /dev/null 2>&1
+
+  if [[ "$?" -ne 0 ]]; then
+    complain "Failed to clone ${kw_repo_url}"
+    if [[ -n "$HOST_KW_CLONED_PATH" ]]; then
+      if is_safe_path_to_remove "$HOST_KW_CLONED_PATH"; then
+        rm -rf "$HOST_KW_CLONED_PATH"
+      else
+        complain "Unsafe path: ${HOST_KW_CLONED_PATH} - Not removing"
+      fi
+    fi
+    exit "$?"
+  fi
+
+  # The VERBOSE variable is set and exported in the run_tests.sh script based
+  # on the command-line options provided by the user. It controls the verbosity
+  # of the output during the test runs.
+  setup_container_environment "$VERBOSE" 'self-update'
+}
+
+# Function to set up an outdated kworkflow environment in a container
+#
+# @container: The name of the container.
+# @kw_cloned_container_path: Temporary directory inside the container where kworkflow is cloned
+# @kw_branch: The branch of kworkflow to be used.
+# @commit_point: The specific commit to be checked out.
+function setup_outdated_kworkflow_environment()
```

Comment:
Just to be sure, the idea is to:

1) in each `_old_version_` test case, install an arbitrary outdated upstream version of kw, using the cloned repo
2) in each `_old_version_` test case, get the expected version from `kw --version` using the the cloned repo
3) in each `_old_version_` test case, run the current version of `kw self-update` (from the local repo actually being tested.

If so, I feel like it is a great approach to circumvent some of the difficulties we mentioned in the offline conversation.

Reply:
Exactly, that's the idea.